tableau_utilities/scripts/connection.py

```diff
@@ -164,10 +173,16 @@ def connection(args, server=None):
         if debugging_logs:
             print(f'EMBEDDING CREDS FOR DATASOURCE ID: {datasource_id}')
 
-        if datasource_id is None:
-            datasource_id = server.get_datasource(datasource_name=args.name, datasource_project=args.project_name).id
+        obj = getattr(server, f'get_datasource')(object_id, object_name, project_name)
+        object_id = obj.id or object_id
+        object_name = obj.name or object_name
+        project_name = obj.project_name or project_name
+
+        print(f'Updating credentials for {object_id} {object_name} {project_name}')
 
-        embed_credential_online(datasource_id, conn_settings, server, debugging_logs)
+        response = embed_credential_online(object_id, conn_settings, server, debugging_logs)
```

Comment:
@JustinGrilli  I can't seem to get embed credentials to work.  It was working before in my testing as I used it last week.  It's not working in this round of testing.

Here's what I did to test.

1. Downloaded a datasource
```
op run --env-file="/Users/jayrosenthal/.prod_env" -- tableau_utilities --name 'Opportunity' --project_name 'Panther All' --include_extract server_operate --download datasource
```

2. Re-published the data source
```
op run --env-file="/Users/jayrosenthal/.prod_env" -- tableau_utilities --file_path /Users/jayrosenthal/code/tableau-utilities/tmp_tdsx_and_config/Opportunity.tdsx --name 'Opportunity' --project_name 'Panther All' server_operate --publish datasource
```

3. Embed Credentials
```
tableau_utilities --location online --name 'Opportunity' --project_name 'Panther All'  connection --connection_operation embed_user_pass
```

4. Here's the response
```
-> % op run --env-file="/Users/jayrosenthal/.prod_env" -- tableau_utilities --location online --name 'Opportunity' --project_name 'Panther All'  connection --connection_operation embed_user_pass
None
Online Datasource, Updating Embedded Username and Password
[32m✅  Connection(id='5661cbf2-d178-4391-9aaf-f7790c88d7da', type=None, datasource_id=None, datasource_name=None, workbook_id=None, workbook_name=None, user_name='TABLEAU_USER', server_address='<concealed by 1Password>-datawarehouse.<concealed by 1Password>computing.com', server_port=None, embed_password=False, password=None, query_tagging_enabled=False)[0m
```

I don't understand why the response has `embed_password=False, ` even though your code appears to set that to `True` in the function.  I also don't understand why I only see a connection_id and no datasource_id in the response.

Are you able to get this function to work?

Reply:
Saw you resolved this already 🙌 

I have not run this, or anything that would alter a datasource online, because at the moment we don't have a staging environment I can test those things in (we recently made some vender changes that caused some changes to tableau online, like loosing our staging env 😢).

But once we get a solution in place to test things online, I will test this and other datasource manipulation, etc.